crates/bevy_ecs/src/system/commands/mod.rs

```diff
@@ -837,30 +837,62 @@ impl<'w, 's, 'a> EntityCommands<'w, 's, 'a> {
         self.commands
     }
 
-    /// Allows querying of an entity's components.
+    /// Allows querying of an entity's components. This is most useful when working with a highly
+    /// composed entity where you wish to modify some of its components.
     ///
-    /// # Panics
-    ///
-    /// The command will panic when applied if the associated entity and/or queried components do not exist.
+    /// If the entity is not available or query fails, then the provided closure will not be executed.
     ///
     /// # Examples
     ///
+    /// ## Conditionally Modify A Component
     /// ```
     /// # use bevy_ecs::prelude::*;
     /// #
     /// # #[derive(Resource)]
     /// # struct PlayerEntity { entity: Entity }
+    /// #
     /// #[derive(Component)]
     /// struct Health(u32);
     ///
-    /// fn zero_health(mut commands: Commands, player: Res<PlayerEntity>) {
+    /// #[derive(Component)]
+    /// struct LuckyStart;
+    ///
+    /// fn spawn_player(mut commands: Commands, player: Res<PlayerEntity>) {
+    ///     commands
+    ///         .entity(player.entity)
+    ///         .insert((Health(100), LuckyStart))
+    ///         /* ... */
+    ///         .with_query::<(&mut Health, &LuckyStart)>(|(mut health, _)| {
+    ///             // Players with the LuckyStart component start with double health.
+    ///             health.0 *= 2;
+    ///         });
+    /// }
+    /// # bevy_ecs::system::assert_is_system(spawn_player);
+    /// ```
+    ///
+    /// ## Modify a Part of a Bundle
+    /// ```
+    /// # use bevy_ecs::prelude::*;
+    /// #
+    /// # #[derive(Component)]
+    /// # struct Transform { translation: f32 }
+    /// # impl Transform {
+    /// #   fn up(&self) -> f32 { 1f32 }
+    /// # }
+    /// #
+    /// # #[derive(Resource)]
+    /// # struct PlayerEntity { entity: Entity }
+    /// #
+    /// fn spawn_player(mut commands: Commands, player: Res<PlayerEntity>) {
     ///     commands
     ///         .entity(player.entity)
-    ///         .with_query::<&mut Health>(|mut health| {
-    ///             health.0 = 0;
+    ///         /* ... */
+    ///         .with_query::<&mut Transform>(|mut transform| {
+    ///             // Players should spawn slightly above their randomly chosen spawn point.
+    ///             transform.translation += 0.1 * transform.up();
```

Comment:
Again, seems like this would be better implemented as a system, instead of adding a closure for every entity where you want to adjust its translation.

Reply:
I do think there's a material difference tho. Adding it as a system permanently adds it to the system schedule, adding whatever overhead (however small) that entails.

I will say that I don't necessarily believe my PR is an absolutely necessary change to Bevy, but it is the minimum reasonable solution (IMO) to the issue raised that this PR is attempting to close. If this PR does get rejected because the feature itself isn't worth adding, I do hope the issue does as well.

Reply:
I agree that this is an important distinction, and really what justifies this feature to me.